src/Prometheus/Collector.php

```diff
@@ -101,4 +97,24 @@ protected function assertLabelsAreDefinedCorrectly(array $labels): void
             throw new InvalidArgumentException(sprintf('Labels are not defined correctly: %s', print_r($labels, true)));
         }
     }
+
+    /**
+     * @param string $metricName
+     */
+    public static function validateMetricName(string $metricName): void
+    {
+        if (preg_match(self::RE_METRIC_LABEL_NAME, $metricName) !== 1) {
+            throw new InvalidArgumentException("Invalid metric name: '" . $metricName . "'");
+        }
+    }
+
+    /**
+     * @param string $label
+     */
+    public static function validateLabel(string $label): void
```

Comment:
I think it would be okay if there is one method which is called `assertMetricNameOrLabelNameIsValid` as they share 100% same logic.

What do you think?

Reply:
I think it makes sense to have two methods here just for having the possibility to change it in the future. Maybe future wise Prometheus will change the requirements for metric names but for labels, they stay the same.